File: src/client-app/src/models/requests/transactions/addIncomeTransactionCommand.ts
export interface AddIncomeTransactionCommand {
    destinationAccountId: string
    senderName: string
    transactionAmount: number
    categoryValue: string
    subcategoryValue: string
    transactionDateTime: Date
}

export class AddIncomeTransactionCommand implements AddIncomeTransactionCommand {
    constructor(values?: AddIncomeTransactionCommand) {
        this.destinationAccountId = values?.destinationAccountId || ''
        this.senderName = values?.senderName || ''
        this.transactionAmount = values?.transactionAmount || 0
        this.categoryValue = values?.categoryValue || ''
        this.subcategoryValue = values?.categoryValue || ''
        this.transactionDateTime = values?.transactionDateTime || new Date()
    }
}

File: src/PennyPlanner.Application/Transactions/AddIncomeTransaction/AddIncomeTransactionCommand.cs
using CommonAbstractions.DB.Messaging;
using PennyPlanner.Domain.Transactions;

namespace PennyPlanner.Application.Transactions.AddIncomeTransaction
{
    public sealed record AddIncomeTransactionCommand(
        string DestinationAccountId,
        string SenderName,
        decimal TransactionAmount,
        string CategoryValue,
        string SubcategoryValue,
        DateTime TransactionDateTime) : ICommand<Transaction>
    {
    }
}
